### ISSUE TITLE
On Linux, alarm will fail if first character in note is a minus

### ISSUE THREAD

**Issue by belzecue** [laurent22/joplin#7457]
## Environment

Joplin version: 2.8.8
Platform: Linux
OS specifics: Linux Mint/Ubuntu 20.04.5 LTS

## Steps to reproduce

1. Load appimage on Ubuntu 20.04.5 LTS
2. Create a new note and on first line create a checkbox with markdown of: - [ ] item
3. Set an alarm on the note for a minute or two in the future, wait, and observe that the notification IS NOT triggered
4. Remove the minus character at the start of the first line
5. Set an alarm on the note for a minute or two in the future, wait, and observe that the notification IS triggered

## Describe what you expected to happen

Linux notifications should trigger for any note with a future alarm time.  This behaviour should not be affected by the contents of a note.  There is a known issue with note content affecting alarms, opened in 2020 and described here [#3925](https://github.com/laurent22/joplin/issues/3925).  Linux notification cannot handle a dash/minus as the first character.  Solution is to escape the first character for any notification string that starts with a minus.

**Comment by Mighil31** [comment]
I noticed that the issue exists when a dash exists in the title as well. But I can't seem to escape the dash when its in the title.

When I run `notify-send "\- [ ] Todo" "\- [ ] Todo"`
This is the notification I get:

![Screenshot from 2022-12-29 10-28-57](https://user-images.githubusercontent.com/47496035/209905622-ae65563e-ad85-46b3-a575-54d493780f4d.png)

The dash in the body is escaped but the one in the title simply has a \ appended. The title in general doesn't look like it gets parsed. For eg when I set an alarm on this note

![Screenshot from 2022-12-29 11-28-20](https://user-images.githubusercontent.com/47496035/209910882-75935398-3482-495b-89f0-081653ce9c30.png)

I get the following notification:

![Screenshot from 2022-12-29 11-28-08](https://user-images.githubusercontent.com/47496035/209910936-317eb728-f33b-434c-bd14-b59c04bde834.png)

The solution then is to simply ignore leading dashes right? Something like this?
```
while(notification.title.length !== 0 && notification.title.charAt(0) === '-')
    notification.title = notification.title.substring(1);
     
```

**Comment by Mighil31** [comment]
Hey @laurent22 Sorry to bug you. I've detailed the problem above. It's a very simple issue that I need some clarification on.

**Comment by laurent22** [comment]
There's two bugs here - first is that all dashes as first characters should be removed, and second is that content should be cleaned up, for example using stripMarkup()

**Comment by XilinJia** [comment]
If no one is working on this, I'd like to give it a shot.

**Comment by laurent22** [comment]
> If no one is working on this, I'd like to give it a shot.

Yes but please do two separate pull requests - one for the dash issue, and one for the encoding issue

**Comment by Mighil31** [comment]
Hey @XilinJia . As you can see above I had already started working on this a bit earlier. Could you please allow me to have a go at it.

**Comment by XilinJia** [comment]
> Hey @XilinJia . As you can see above I had already started working on this a bit earlier. Could you please allow me to have a go at it.

Sure @Mighil31.  Good luck
